src/extension.ts

```diff
@@ -24,9 +25,45 @@ export async function activate(context: vscode.ExtensionContext) {
                     await activateConfigurePipeline();
             },
             TelemetryKeys.ExtensionActivationDuration);
-    });
+	});
+	
+    // The YAML language server is implemented in node
+	let serverModule = context.asAbsolutePath(path.join('node_modules', 'yaml-language-server', 'out', 'server', 'src', 'server.js'));
+
+	// The debug options for the server
+	let debugOptions = { execArgv: ['--nolazy', '--inspect=6009'] };
 
-    logger.log('Extension has been activated!', 'ExtensionActivated');
+	// If the extension is launched in debug mode then the debug server options are used
+	// Otherwise the run options are used
+	let serverOptions: ServerOptions = {
+		run : { module: serverModule, transport: TransportKind.ipc },
+		debug: { module: serverModule, transport: TransportKind.ipc, options: debugOptions }
+	};
+
+	// Options to control the scope of language client
+	let clientOptions: LanguageClientOptions = {
+		documentSelector: [
+			{language: "yaml", pattern: "**/.github/workflows/**", scheme: "file"}
+		],
+		synchronize: {
+			// Synchronize these setting sections with the server
```

Comment:
should we lock onto a fixed version of server? otherwise how will we know that we have diverged from server?

Reply:
what I meant was whether we should use a fixed version instead of `^` in `"yaml-language-server": "^0.8.0"`

Reply:
It won't be required as `try-catch` block has already been added, in case our client becomes incompatible with server in future.